libs/deck-layers/src/layers/rulers/RulersLayer.ts

```diff
@@ -0,0 +1,91 @@
+import { GeoJsonLayerProps, GeoJsonLayer, TextLayer } from '@deck.gl/layers/typed'
+import { CompositeLayer, LayersList } from '@deck.gl/core/typed'
+import { PathStyleExtension } from '@deck.gl/extensions'
+import { Position } from '@deck.gl/core'
+import { rhumbBearing } from '@turf/turf'
+import { Ruler } from '@globalfishingwatch/layer-composer'
+import { LayerGroup, getLayerGroupOffset } from '../../utils'
+import {
+  getGreatCircleMultiLine,
+  getRulerLengthLabel,
+  getRulerStartAndEndPoints,
+} from './rulers.utils'
+import { COLOR } from './rulers.config'
+
+type RulersLayerProps = GeoJsonLayerProps & {
+  rulers: Ruler[]
+}
+
+type RulerLabel = {
+  text: string
+  position: Position
+  bearing: number
+}
+
+const getFeaturesFromRulers = (rulers: Ruler[]) => {
+  return rulers.flatMap((ruler: Ruler) => [
+    getGreatCircleMultiLine(ruler),
+    ...getRulerStartAndEndPoints(ruler),
+  ])
+}
+export class RulersLayer extends CompositeLayer<RulersLayerProps> {
+  static layerName = 'RulersLayer'
+  layers: LayersList = []
+  updateState({ props }: { props: RulersLayerProps }) {
+    const labels = props.rulers.map((ruler: Ruler) => {
+      const line = getGreatCircleMultiLine(ruler)
+      const centerIndex = Math.round(line.geometry.coordinates.length / 2)
+      const centerPoint = line.geometry.coordinates[centerIndex]
+      const anchorPoints = line.geometry.coordinates.slice(centerIndex, centerIndex + 2)
+      const bearing = rhumbBearing(anchorPoints[0] as Position, anchorPoints[1] as Position)
+      return {
+        text: getRulerLengthLabel(ruler),
+        position: centerPoint,
+        bearing: bearing <= 0 ? 270 - bearing : 90 - bearing,
+      }
+    })
+    this.setState({ labels })
+  }
+  renderLayers() {
+    const { rulers, visible } = this.props
+    this.layers = [
```

Comment:
Do we need to create a layers reference on the class? why not just returning the LayersList array directly?